# Task
Analyze a dataset by identifying numerical and categorical columns, calculating descriptive statistics for each type, and then providing a comprehensive summary and overview of the dataset.

## Cargar Dataset

### Subtask:
Cargar un dataset de ejemplo de una biblioteca de Python (por ejemplo, el dataset 'tips' de seaborn) en un DataFrame de pandas.


**Reasoning**:
First, I'll import the necessary libraries, pandas and seaborn, then load the 'tips' dataset into a pandas DataFrame named `df` as instructed.



In [ ]:
import pandas as pd
import seaborn as sns

# Load the 'tips' dataset from seaborn
df = sns.load_dataset('tips')

# Display the first few rows of the DataFrame to verify
print("Dataset 'tips' loaded successfully into DataFrame 'df'.")
df.head()

Dataset 'tips' loaded successfully into DataFrame 'df'.


,total_bill,tip,sex,smoker,day,time,size
0,16.99,1.01,Female,No,Sun,Dinner,2
1,10.34,1.66,Male,No,Sun,Dinner,3
2,21.01,3.50,Male,No,Sun,Dinner,3
3,23.68,3.31,Male,No,Sun,Dinner,2
4,24.59,3.61,Female,No,Sun,Dinner,4


## Identificar Columnas

### Subtask:
Identificar y separar las columnas numéricas y categóricas del DataFrame cargado.


**Reasoning**:
To identify and separate numerical and categorical columns, I will use the `select_dtypes` method from pandas, first including numerical types, then including object and category types, and finally print the identified column names.



In [ ]:
numerical_cols = df.select_dtypes(include=['number']).columns.tolist()
categorical_cols = df.select_dtypes(include=['object', 'category']).columns.tolist()

print(f"Columnas Numéricas: {numerical_cols}")
print(f"Columnas Categóricas: {categorical_cols}")

Columnas Numéricas: ['total_bill', 'tip', 'size']
Columnas Categóricas: ['sex', 'smoker', 'day', 'time']


## Estadísticas Numéricas

### Subtask:
Calcular y mostrar las estadísticas descriptivas (como media, desviación estándar, mínimo, máximo, cuartiles) solo para las columnas numéricas.


**Reasoning**:
To calculate the descriptive statistics for the numerical columns, I will select these columns from the DataFrame `df` using the `numerical_cols` list and then apply the `.describe()` method.



In [ ]:
descriptive_stats_numerical = df[numerical_cols].describe()
print("Estadísticas Descriptivas para Columnas Numéricas:")
print(descriptive_stats_numerical)

Estadísticas Descriptivas para Columnas Numéricas:
       total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000


## Estadísticas Categóricas

### Subtask:
Calcular y mostrar las estadísticas descriptivas (como conteo, número de valores únicos, valor más frecuente, frecuencia del valor más frecuente) solo para las columnas categóricas.


**Reasoning**:
To calculate the descriptive statistics for the categorical columns, I will select these columns from the DataFrame `df` using the `categorical_cols` list and then apply the `.describe()` method, printing the result with a descriptive header.



In [ ]:
descriptive_stats_categorical = df[categorical_cols].describe()
print("Estadísticas Descriptivas para Columnas Categóricas:")
descriptive_stats_categorical

Estadísticas Descriptivas para Columnas Categóricas:


,sex,smoker,day,time
count,244,244,244,244
unique,2,2,4,2
top,Male,No,Sat,Dinner
freq,157,151,87,176


## Estadísticas Completas

### Subtask:
Calcular y mostrar las estadísticas descriptivas para todas las columnas del dataset (numéricas y categóricas).


**Reasoning**:
To calculate descriptive statistics for all columns, including both numerical and categorical, I will apply the `.describe(include='all')` method directly to the DataFrame `df` and then print the result with a clear header.



In [ ]:
overall_descriptive_stats = df.describe(include='all')
print("Estadísticas Descriptivas para TODAS las Columnas:")
overall_descriptive_stats

Estadísticas Descriptivas para TODAS las Columnas:


,total_bill,tip,sex,smoker,day,time,size
count,244.000000,244.000000,244,244,244,244,244.000000
unique,NaN,NaN,2,2,4,2,NaN
top,NaN,NaN,Male,No,Sat,Dinner,NaN
freq,NaN,NaN,157,151,87,176,NaN
mean,19.785943,2.998279,NaN,NaN,NaN,NaN,2.569672
std,8.902412,1.383638,NaN,NaN,NaN,NaN,0.951100
min,3.070000,1.000000,NaN,NaN,NaN,NaN,1.000000
25%,13.347500,2.000000,NaN,NaN,NaN,NaN,2.000000
50%,17.795000,2.900000,NaN,NaN,NaN,NaN,2.000000
75%,24.127500,3.562500,NaN,NaN,NaN,NaN,3.000000


## Final Task

### Subtask:
Resumir las estadísticas descriptivas obtenidas para las columnas numéricas y categóricas, y proporcionar una visión general del dataset.


### Tablas de Frecuencia para Columnas Categóricas

## Summary:

### Data Analysis Key Findings

*   **Dataset Overview**: The analysis was performed on the 'tips' dataset, which contains 244 entries.
*   **Column Identification**:
    *   **Numerical Columns**: `total_bill`, `tip`, `size` were identified.
    *   **Categorical Columns**: `sex`, `smoker`, `day`, `time` were identified.
*   **Numerical Statistics**:
    *   **`total_bill`**: Average bill was approximately \$19.79 (standard deviation \$8.90), ranging from \$3.07 to \$50.81.
    *   **`tip`**: Average tip was around \$3.00 (standard deviation \$1.38), ranging from \$1.00 to \$10.00.
    *   **`size`**: The average party size was approximately 2.57 individuals (standard deviation 0.95), ranging from 1 to 6.
*   **Categorical Statistics**:
    *   **`sex`**: Comprises 2 unique values, with 'Male' being the most frequent (157 occurrences).
    *   **`smoker`**: Comprises 2 unique values, with 'No' being the most frequent (151 occurrences).
    *   **`day`**: Comprises 4 unique values, with 'Sat' being the most frequent (87 occurrences).
    *   **`time`**: Comprises 2 unique values, with 'Dinner' being the most frequent (176 occurrences).
*   **Overall Summary**: The dataset includes a mix of numerical and categorical variables, providing information on dining bills, tips, party sizes, and various contextual factors like gender, smoking preference, day of the week, and mealtime.

### Insights or Next Steps

*   **Insight**: The 'tips' dataset reveals that dinner service on Saturdays tends to be the busiest, with male customers and non-smokers being the most frequent patrons. Tips generally average around 15% of the total bill based on the average tip (\$3.00) and average total bill (\$19.79), although further analysis would confirm this ratio more precisely.
*   **Next Steps**: Investigate the relationship between `total_bill` and `tip` to understand tipping habits, potentially exploring how categorical variables like `sex`, `smoker`, `day`, and `time` influence tip percentages or total bill amounts.


In [ ]:
for col in categorical_cols:
    print(f"\n### Tabla de Frecuencias para '{col}':")
    absolute_freq = df[col].value_counts().rename('Frecuencia Absoluta')
    relative_freq = (df[col].value_counts(normalize=True) * 100).rename('Frecuencia Relativa (%)')

    # Combine into a single DataFrame for display
    frequency_table = pd.concat([absolute_freq, relative_freq], axis=1)
    display(frequency_table)
    print("-----------------------------------")


### Tabla de Frecuencias para 'sex':


,Frecuencia Absoluta,Frecuencia Relativa (%)
sex,,
Male,157,64.344262
Female,87,35.655738


-----------------------------------

### Tabla de Frecuencias para 'smoker':


,Frecuencia Absoluta,Frecuencia Relativa (%)
smoker,,
No,151,61.885246
Yes,93,38.114754


-----------------------------------

### Tabla de Frecuencias para 'day':


,Frecuencia Absoluta,Frecuencia Relativa (%)
day,,
Sat,87,35.655738
Sun,76,31.147541
Thur,62,25.409836
Fri,19,7.786885


-----------------------------------

### Tabla de Frecuencias para 'time':


,Frecuencia Absoluta,Frecuencia Relativa (%)
time,,
Dinner,176,72.131148
Lunch,68,27.868852


-----------------------------------
